From 0152c299238ddb4c200b1f58b376556053d7de09 Mon Sep 17 00:00:00 2001
From: "deepak.s" <deepak.s@adlinktech.com>
Date: Mon, 15 Jul 2024 14:26:04 +0530
Subject: [PATCH 2/3] u-boot-modify-Kconfig-Makefile-to-build-lec-imx8mm

---
 arch/arm/mach-imx/imx8m/Kconfig | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/mach-imx/imx8m/Kconfig b/arch/arm/mach-imx/imx8m/Kconfig
index 473cfa8e5e39..2cc6147e9ba7 100644
--- a/arch/arm/mach-imx/imx8m/Kconfig
+++ b/arch/arm/mach-imx/imx8m/Kconfig
@@ -75,6 +75,12 @@ config TARGET_IMX8MQ_CM
 	select IMX8MQ
 	select IMX8M_LPDDR4
 
+config TARGET_LEC_IMX8MM
+	bool "Adlink lec-imx8mm LPDDR4 board"
+	select IMX8MM
+	select SUPPORT_SPL
+	select IMX8M_LPDDR4
+
 config TARGET_IMX8MQ_EVK
 	bool "imx8mq_evk"
 	select IMX8MQ
@@ -482,5 +488,6 @@ source "board/variscite/imx8mn_var_som/Kconfig"
 source "board/toradex/verdin-imx8mm/Kconfig"
 source "board/toradex/verdin-imx8mp/Kconfig"
 source "board/voxelbotics/navqp/Kconfig"
+source "board/adlink/lec-imx8mm/Kconfig"
 
 endif
-- 
2.25.1

