From 909f198be31f5412b5c945fbc47ddd84e522656d Mon Sep 17 00:00:00 2001
From: "deepak.s" <deepak.s@adlinktech.com>
Date: Mon, 29 Jul 2024 13:15:53 +0530
Subject: [PATCH] Add-FW-action-hotplug

---
 bt/bt_drv.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bt/bt_drv.h b/bt/bt_drv.h
index f02af2b..3efb01f 100755
--- a/bt/bt_drv.h
+++ b/bt/bt_drv.h
@@ -32,7 +32,7 @@
 
 /** MAX adapter BT driver supported */
 #define MAX_BT_ADAPTER    3
-
+#define FW_ACTION_HOTPLUG 1
 #ifndef BIT
 /** BIT definition */
 #define BIT(x) (1UL << (x))
-- 
2.43.2

