From 974213a433e80236c911d73d9598a8ad041b595e Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Mon, 17 Aug 2020 18:48:30 +0530
Subject: [PATCH 40/40] Critical and passive temperature is 103*c

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 drivers/thermal/imx_thermal.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/thermal/imx_thermal.c b/drivers/thermal/imx_thermal.c
index 1e5eec1a7f0f..aeab153f36a8 100644
--- a/drivers/thermal/imx_thermal.c
+++ b/drivers/thermal/imx_thermal.c
@@ -590,11 +590,11 @@ static void imx_init_temp_grade(struct platform_device *pdev, u32 ocotp_mem0)
 	}
 
 	/*
-	 * Set the critical trip point at 5 °C under max
-	 * Set the passive trip point at 10 °C under max (changeable via sysfs)
+	 * Set the critical trip point at 2 °C under max
+	 * Set the passive trip point at 2 °C under max (changeable via sysfs)
 	 */
-	data->temp_critical = data->temp_max - (1000 * 5);
-	data->temp_passive = data->temp_max - (1000 * 10);
+	data->temp_critical = data->temp_max - (1000 * 2);
+	data->temp_passive = data->temp_max - (1000 * 2);
 }
 
 static int imx_init_from_tempmon_data(struct platform_device *pdev)
-- 
2.17.1

