From 696be0a878beb2c060fdfc9923023c4296794208 Mon Sep 17 00:00:00 2001
From: Dinesh kumar <dineshkumar.varadarajan@ltts.com>
Date: Tue, 26 Nov 2019 13:06:14 +0530
Subject: [PATCH] Critical and passive temperature is 103*c

---
 drivers/thermal/imx_thermal.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/thermal/imx_thermal.c b/drivers/thermal/imx_thermal.c
index deb2d16..5c9832f 100644
--- a/drivers/thermal/imx_thermal.c
+++ b/drivers/thermal/imx_thermal.c
@@ -96,6 +96,13 @@ enum imx_thermal_trip {
 
 #define IMX_TEMP_PASSIVE_COOL_DELTA	10000
 
+/*
+ * The maximum die temperature on i.MX parts is 105 C, let's give some cushion
+ * for noise and possible temperature rise between measurements.
+ */
+#define IMX_TEMP_CRITICAL		103000
+
+
 #define IMX_POLLING_DELAY		2000 /* millisecond */
 #define IMX_PASSIVE_DELAY		1000
 
@@ -661,8 +668,8 @@ static int imx_get_sensor_data(struct platform_device *pdev)
 	 * Set the critical trip point at 5C under max
 	 * Set the passive trip point at 10C under max (can change via sysfs)
 	 */
-	data->temp_critical = data->temp_max - (1000 * 5);
-	data->temp_passive = data->temp_max - (1000 * 10);
+	data->temp_critical = data->temp_max - (1000 * 2);
+	data->temp_passive = data->temp_max - (1000 * 2);
 
 	return 0;
 }
-- 
1.9.1

