From 75f9d33629be50ce00ab97e1006d76eae4275473 Mon Sep 17 00:00:00 2001
From: Dinesh kumar <dinesh.v@adlinktech.com>
Date: Sun, 10 Nov 2024 15:08:24 +0530
Subject: [PATCH] Added DDR config binary into Image

---
 iMX95/soc.mak | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/iMX95/soc.mak b/iMX95/soc.mak
index 4cce38e..6fd549d 100644
--- a/iMX95/soc.mak
+++ b/iMX95/soc.mak
@@ -64,12 +64,14 @@ OEI_A55_LOAD_ADDR ?= 0x20498000
 OEI_A55_ENTR_ADDR ?= $(OEI_A55_LOAD_ADDR)
 OEI_M33_LOAD_ADDR ?= 0x1ffc0000
 OEI_M33_ENTR_ADDR ?= 0x1ffc0001	# = real entry address (0x1ffc0000) + 1
+OEI_M33_DAT_LOAD_ADDR ?= 0x4aa80000
 
 OEI_OPT_A55 ?=
 OEI_OPT_M33 ?=
 
 OEI_IMG_A55 ?=
 OEI_IMG_M33 ?=
+OEI_OPT_M33_DATA ?=
 
 lpddr_imem = $(LPDDR_TYPE)_imem$(LPDDR_FW_VERSION).bin
 lpddr_dmem = $(LPDDR_TYPE)_dmem$(LPDDR_FW_VERSION).bin
@@ -81,6 +83,7 @@ OEI_A55_DDR_IMG ?= oei-a55-ddr.bin
 OEI_M33_DDR_IMG ?= oei-m33-ddr.bin
 OEI_A55_TCM_IMG ?= oei-a55-tcm.bin
 OEI_M33_TCM_IMG ?= oei-m33-tcm.bin
+OEI_M33_DAT_IMG ?= oei-m33-dat.bin
 
 A55_OEI_DDRFW = a55-oei-ddrfw.bin
 M33_OEI_DDRFW = m33-oei-ddrfw.bin
@@ -111,6 +114,11 @@ OEI_OPT_M33 += -oei $(OEI_M33_TCM_IMG) m33 $(OEI_M33_ENTR_ADDR) $(OEI_M33_LOAD_A
 OEI_IMG_M33 += $(OEI_M33_TCM_IMG)
 endif
 
+ifneq (,$(wildcard $(OEI_M33_DAT_IMG)))
+OEI_OPT_M33_DATA += -data $(OEI_M33_DAT_IMG) m33 $(OEI_M33_DAT_LOAD_ADDR)
+OEI_IMG_M33 += $(OEI_M33_DAT_IMG)
+endif
+
 ifeq (,$(OEI_IMG_M33))
 $(warning "Note: There are no Cortex-M33 oei images")
 endif
@@ -305,7 +313,7 @@ flash_a55_no_ahabfw: $(MKIMG) $(MCU_IMG) u-boot-atf-container.img $(SPL_A55_IMG)
 	$(call append_container,u-boot-atf-container.img,1)
 
 flash_a55_flexspi: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(SPL_A55_IMG) $(OEI_IMG_M33) fcb.bin u-boot-atf-container.img
-	./$(MKIMG) -soc IMX9 -dev flexspi -append $(AHAB_IMG) -c $(OEI_OPT_M33) -msel $(MSEL) \
+	./$(MKIMG) -soc IMX9 -dev flexspi -append $(AHAB_IMG) -c $(OEI_OPT_M33_DATA) $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
 		   -ap $(SPL_A55_IMG) a55 $(SPL_LOAD_ADDR_M33_VIEW) $(V2X_DUMMY) -fcb fcb.bin $(FCB_LOAD_ADDR) -out flash.bin
 	$(call append_container,u-boot-atf-container.img,1)
@@ -342,7 +350,7 @@ flash_m7_ddr: $(MKIMG) $(MCU_IMG) $(M7_IMG) $(OEI_IMG_M33) $(AHAB_IMG)
 		   -m7 $(M7_IMG) 0 $(M7_DDR_ADDR) $(M7_DDR_ADDR) $(V2X_DUMMY) -out flash.bin
 
 flash_all: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M7_IMG) u-boot-atf-container.img $(SPL_A55_IMG) $(OEI_IMG_M33)
-	./$(MKIMG) -soc IMX9 -append $(AHAB_IMG) -c $(OEI_OPT_M33) -msel $(MSEL) \
+	./$(MKIMG) -soc IMX9 -append $(AHAB_IMG) -c $(OEI_OPT_M33_DATA) $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
 		   -m7 $(M7_IMG) 0 $(M7_TCM_ADDR) $(M7_TCM_ADDR_ALIAS)  \
 		   -ap $(SPL_A55_IMG) a55 $(SPL_LOAD_ADDR_M33_VIEW) $(V2X_DUMMY) -out flash.bin
-- 
2.25.1

